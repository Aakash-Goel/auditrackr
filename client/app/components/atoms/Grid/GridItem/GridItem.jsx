import React from 'react';
import { object, node, string } from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import Grid from '@material-ui/core/Grid';

/**
 * Type checking - Define prop types
 * @private
 */
const propTypes = {
  classes: object.isRequired,
  children: node.isRequired,
  className: string,
};

/**
 * Define default prop values
 * @private
 */
const defaultProps = {
  className: '',
};

const style = {
  grid: {
    position: 'relative',
    width: '100%',
    minHeight: '1px',
    paddingRight: '15px',
    paddingLeft: '15px',
    flexBasis: 'auto',
  },
};

const GridItem = ({ ...props }) => {
  const { classes, children, className, ...rest } = props;
  return (
    <Grid item {...rest} className={`${classes.grid} ${className}`}>
      {children}
    </Grid>
  );
};

GridItem.propTypes = propTypes;
GridItem.defaultProps = defaultProps;

export default withStyles(style)(GridItem);
