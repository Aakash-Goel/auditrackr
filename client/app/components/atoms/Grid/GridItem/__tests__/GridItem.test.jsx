import React from 'react';
import TestRenderer from 'react-test-renderer';

import GridItem from '../GridItem';

describe('<GridItem/>', () => {
  it('renders correctly with empty props', () => {
    const gridItemWrapper = TestRenderer.create(
      <GridItem>My content goes here</GridItem>
    ).toJSON();
    expect(gridItemWrapper).toMatchSnapshot();
  });
});
