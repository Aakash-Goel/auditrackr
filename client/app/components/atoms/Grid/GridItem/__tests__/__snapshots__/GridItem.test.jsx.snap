// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItem/> renders correctly with empty props 1`] = `
<div
  className="MuiGrid-root GridItem-grid-1 MuiGrid-item"
>
  My content goes here
</div>
`;
